plugins/BEdita/Core/src/Model/Table/DateRangesTable.php

```diff
@@ -97,4 +95,52 @@ public function buildRules(RulesChecker $rules)
 
         return $rules;
     }
+
+    /**
+     * Find objects by date range.
+     *
+     * Create a query to filter objects using start and end date conditions.
+     * Accepted options are:
+     *   - 'start_date' or 'end_date'
+     *   - mandatory sub-option must be one of 'gt' (greather than), 'lt' (less than),
+     *          'ge' (greater or equal), 'le' (less or equal) with a date
+     *
+     * Examples
+     * ```
+     * // find events with a start date after '2017-03-01'
+     * $table->find('dateRanges', ['start_date' => ['gt' => '2017-03-01']], 'Events');
+     *
+     * // find events with an ending date before '2017-05-01 22:00:00'
+     * $table->find('dateRanges', ['end_date' => ['lt' => '2017-05-01 22:00:00']], 'Events');
+     * ```
+     *
+     * @param \Cake\ORM\Query $query Query object instance.
+     * @param array $options Array of acceptable date range conditions.
+     * @param string $objectAlias Name of object type to filter like 'Events'.
+     * @return \Cake\ORM\Query
+     */
+    public function findDateRanges(Query $query, array $options, $objectAlias = 'Objects')
+    {
+        $options = array_intersect_key($options, ['start_date' => 0, 'end_date' => 0]);
+        if (!empty($options)) {
+            $conditions = [];
+            $subopts = ['gt' => '>', 'lt' => '<', 'ge' => '>=', 'le' => '<='];
+            foreach ($options as $key => $value) {
+                if (is_array($value) && ($k = key($value)) && !empty($subopts[$k])) {
+                    $conditions[] = sprintf("%s.%s %s '%s'", $this->getAlias(), $key, $subopts[$k], $value[$k]);
```

Comment:
Building conditions this way is highly discouraged, AFAIK, as it opens up a way for SQL injections…
It should probably be something like:

```php
$cond = sprintf("%s.%s %s", $this->getAlias(), $key, $subopts[$k]);
$conditions[$cond] = $value[$k];
```

Or, perhaps even better, use `QueryExpression` instead… 🤔